### PR TITLE
Mc enter event

### PR DESCRIPTION
**Pull request by MCrank**
I got on a roll and added some additional things that should have gone into their own card  😞 

* Updated font-awesome to latest and adjusted styling to match new version
* Added "Enter" key event handler import into main.js
* Added validation for empty message value
* Update message output format from boilerplate

### CODE REVIEW

**Comment by MCrank** [comment]
@m-khezri @mjleavell @lancehamilton24  Hold off on this.  I need to fix a real nasty bug I just found on the user drop down where we renamed the id.

**Comment by MCrank** [comment]
* Added navBarUserSelectEvents
* Updated enterKeyMsgEvent to handle validation for both input and userSelect
* Cleaned up userPopulate string builder to enabe better event handling
* Updated inputValidation in utils.js to handle input and select validation rules

**Comment by MCrank** [comment]
@mjleavell @m-khezri @lancehamilton24  Okay I think I got it all working. This is a much bigger PR than it should have been but I think it provides some nice feeback.  Let me know if you have any questions.

File: index.html [hunk 1]
```diff
@@ -16,14 +17,19 @@
...
+                </select>
+
             </div>
         </div>
-        <button type="button" class="btn btn-outline-info my-auto" id="theme-button">Select Theme <i class="fa fa-eyedropper"></i>
-        </button>
+        <button type="button" class="btn btn-outline-info my-auto" id="themeButton">Select Theme <i class="fas fa-eye-dropper"></i></button>
     </nav>
+
+
```
Comment (mjleavell) [inline index.html]: delete line 32

File: index.html [hunk 2]
```diff
@@ -16,14 +17,19 @@
...
         <div class="input-group mx-5 my-auto">
-            <input type="text" placeholder="Enter your message" class="form-control" aria-label="Text input with dropdown button" id="message-input">
+            <input type="text" placeholder="Enter your message" class="form-control" aria-label="Text input with dropdown button"
+                id="message-input">
             <div class="input-group-append" id="users-select">
-                <!-- Populate users here from an array -->
+                <select class="selectpicker btn-success" id="user-selected">
+                    <!-- Populate users here from an array -->
+                </select>
+
```
Comment (mjleavell) [inline index.html]: delete line 26

File: index.html [hunk 3]
```diff
@@ -35,8 +41,10 @@ <h3 class="mb-0">Messages:</h3>
                 </div>
             </div>
         </div>
-        <div class="message-container bg-light mx-auto p-5 rounded shadow" id="message-output"></div>
+        <!-- <div class="message-container bg-light mx-auto p-5 rounded shadow" id="message-output"></div> -->
     </div>
+    <!-- <div class="bg-light rounded w-75 mx-auto shadow" id="message-output"></div> -->
+    <div class="w-75 mx-auto bg-light rounded shadow pt-2" id="message-output"></div>
```
Comment (mjleavell) [inline index.html]: delete line 49

**Review by mjleavell — COMMENTED** [review]
Looks awesome! Once these minor changes are made, i'm good with it.

File: index.html (hunk 1 above)
Comment (MCrank) [reply index.html]: Done

File: index.html (hunk 2 above)
Comment (MCrank) [reply index.html]: Done

File: index.html (hunk 3 above)
Comment (MCrank) [reply index.html]: Done

**Comment by mjleavell** [comment]
👍 🦄 🎉

**Pull request merged**